week2-1/main.js

```diff
@@ -0,0 +1,121 @@
+import pic1 from './assets/김규민.jpeg';
+import pic2 from './assets/전희선.jpeg';
+import pic3 from './assets/서혜은.jpg';
+import pic4 from './assets/황주희.jpeg';
+import pic5 from './assets/백지연.png';
+
+const $ = (selector) => document.querySelector(selector);
+
+let currentStep = 0;
+
+const quizList = [
+  {
+    src: pic1,
+    answer: '김규민',
+  },
+  {
+    src: pic2,
+    answer: '전희선',
+  },
+  {
+    src: pic3,
+    answer: '서혜은',
+  },
+  {
+    src: pic4,
+    answer: '황주희',
+  },
+  {
+    src: pic5,
+    answer: '백지연',
+  },
+];
+
+function showModal(sentence, keep) {
+  const modal = $('.modal');
+  const modalBody = $('p.modal__body');
+
+  modalBody.innerHTML = sentence;
+  modal.classList.remove('hide');
+
+  noShowModalEvent(modal);
+
+  if (keep) return;
+
+  setTimeout(function () {
+    modal.classList.add('hide');
+  }, 1000);
+}
+
+function noShowModalEvent(modal) {
+  modal.addEventListener('click', function (e) {
+    modal.classList.add('hide');
+  });
+}
+
+function goNextLevel(score, image) {
+  currentStep++;
+  const scoreBoard = $('.scoreBoard');
+  scoreBoard.animate([{ transform: 'scaleY(2)' }, { transform: 'scaleY(1)' }], {
+    duration: 2000,
+  });
+  score.innerHTML = +score.innerHTML + 1;
+
+  if (currentStep === quizList.length) {
+    showModal(`<a href="/">메인으로</a>`, true);
+    return;
+  }
+
+  showImg(image);
+}
+
+function showImg(image) {
+  showModal('이미지 로딩 중 입니다 :)', true);
+  image.src = quizList[currentStep].src;
+  image.addEventListener('load', function (e) {
+    $('.modal').classList.add('hide');
+  });
+}
+
+function initGame({ score, image }) {
+  currentStep = 0;
+  score.innerText = 0;
+  showImg(image);
+}
+
+function attachAnswerEvent({ score, answer, image }) {
+  answer.addEventListener('click', function (e) {
+    if (e.target instanceof HTMLElement) {
```

Comment:
instanceof 왜 해줬는지 궁금해요!

Reply:
밑에 `innerHTML` 프로퍼티를 가져오기 위해 narrowing 해준 게 아닐까요!?

Reply:
@joohaem 그렇다기엔 `HTMLElement`로 내로잉 해줘가지구 근데 아마 `HTMLLIElement`로 해주려던 것 같기도 해여

Reply:
그게 훨씬 정확하겠군요 👍👍

```suggestion
    if (e.target instanceof HTMLLIElement) {
```

아마 `HTMLElement`를 사용하면 **버튼 사이 빈 공간 클릭 시**에도
narrowing 되는 발생하는 문제가 생기겠네욥
틀렸습니다 모달 뜨려나,?

Reply:
맞아요! HTMLLIElement가 맞네요! 버튼 사이 빈 공간 클릭하면 틀렸습니다 모달이 뜹니다! 감사합니다 👍